_code/InSituDiff/vignettes/InSituDiff basic vignette.Rmd

```diff
@@ -0,0 +1,220 @@
+
+InSituDiff works by comparing cellular neighborhoods in disease tissues to matching control neighborhoods. 
+This means you can pass it a study's complete data, regardless of how diverse the disease conditions are,
+so long as the control tissues are relevant to all disease tissues.
+If your controls are relevant to different tissues (e.g. healthy liver samples as controls for liver disease
+ and healthy kidney controls for kidney disease), then run your data in batches according to which control tissues are relevant. 
+
+## Usage
+
+Start by loading the package and necessary data, a handful of FOVs from kidneys,
+ one with childhood lupus and two controls:
+
+```{r prep}
+library(InSituDiff)
+library(pheatmap)
+# load demo data from lupus nephritis:
+data(minisle)
+# normalize the counts matrix (InSituDiff is best run on normalized data):
+normconstants <- pmax(Matrix::rowSums(minisle$counts), 20) 
+norm_factors <- Matrix::Diagonal(x = 1 / normconstants)
+minisle$norm <- norm_factors %*% minisle$counts * mean(normconstants)
+rownames(minisle$norm) <- rownames(minisle$counts)
+```
+
+## Setting up InSituDiff
+
+Now we set ourselves up for calculating perturbation scores by running ```initializeISD```.
+This function defines two mappings needed for perturbation scoring: 
+
+1. The cells making up each cell's neighborhood
+2. For each cell, the identify of the control well with the most-similar neighborhood
+
+```{r initialize}
+obj <- initializeISD(
+  mat = minisle$norm, # matrix of normalized expression
+  xy = minisle$xy, # matrix of xy coordinates
+  tissue = minisle$annot$tissue, # vector of cells' tissue IDs
+  iscontrol = grepl("control", minisle$annot$tissue), 
+  k = 50) # define cellular neighborhoods as a cell's 50 nearest neighbors
+str(obj)
+```
+
+And that's all we need to start analysis.  
+
+## Calculating neighborhood perturbations
+
+Below we'll show three useful analyses of perturbations we can perform. 
+But first, at any time you like, you can easily compute a matrix of perturbation scores
+ for any cells and genes you like. 
+To get perturbations for all cells and genes, just enter ```cells = NULL, genes = NULL```.
+*Warning*: for larger datasets, attempting to compute this matrix for all cells and all genes risks overwhelming your R session's memory.
+
+```{r getdata}
+selectedperturbations <- getPerturbations(
+  x = minisle$norm, 
+  obj = obj, 
+  cells = rownames(minisle$counts)[minisle$annot$tissue == "SP19_1139"], # only look at one tissue
+  genes = c("IGHA1", "FGFR3", "HIF1A", "ACE2", "SAT1"),  # only look at a few genes of interest
+  residtype = "log2ratio",  # look at perturbations on the scale of log fold-changes
+  eps = 1)
```

Comment:
eps has a big enough impact that I'd recommend a short commented description here